src/api/functional_tests.js

```diff
@@ -1,306 +1,272 @@
+  let return_data = [];
+
+  try {
+    const { data } = await axios.get(workflowCheckRunAnnotationApi);
+    for (let i = 0; i < data.length; i++) {
+      const run_annotation = data[i];
+      const annotation = {
+        message: run_annotation.message,
+        annotation_level: run_annotation.annotation_level,
+        start_line: run_annotation.start_line,
+        end_line: run_annotation.end_line,
+      };
+
+      return_data.push(annotation);
     }
-    
-    return return_data;
+  } catch (err) {
+    console.log("Unable to fetch _getWorkflowCheckRunAnnotation", err);
+    return;
+  }
+
+  return return_data;
 }
 
 /// full api
 export const getFullTestWorkflowReport = async () => {
-    console.log('inside line 123');
-    let return_data = {        
-        fetched: moment().format().valueOf(),        
-        workflow_runs_data: []
-    };
-
-    if(await _getLocalData()){
-        console.log('line 130');        
-        const localData = await _getDataFromLocal()
-        return_data.fetch_message = "Its too early to request again, clear expiretime";
-        return_data.workflow_runs_data = localData.workflow_runs_data
-        return return_data;
-    }
+  let return_data = {
+    fetched: moment().format().valueOf(),
+    workflow_runs_data: [],
+  };
+
+  if (await _getLocalData()) {
+    const localData = await _getDataFromLocal();
+    return_data.fetch_message =
+      "Its too early to request again, clear expiretime";
+    return_data.workflow_runs_data = localData.workflow_runs_data;
+    return return_data;
+  }
 
-    // console.log('line 136');
-    // if(await _checkForUpdate()){
-    //     console.log('line 139');
-    //     const localData = await _getDataFromLocal() 
-    //     return_data.fetch_message = "No New Workflows Available";
-    //     return_data.workflow_runs_data = localData.workflow_runs_data
-    //     return return_data;
-    // }
-    
-    console.log('line 144');
-    try {
-        console.log('new updates detected get new data');
-        // get latest workflow runs for main branch
-        const workflows = await _getSingleWorkflow(5937682)
-        // get checks (functional test matrix run) for each workflow run
-        for(let i=0; i < workflows.length; i++){
-            let workflow = {
-                workflow_run: {},
-                test_runs: [],                
-                passed_count: 0,
-                failed_count: 0,
-                flaked_count: 0
+  if (await _checkForUpdate()) {
+    console.log("line 139");
+    const localData = await _getDataFromLocal();
+    return_data.fetch_message = "No New Workflows Available";
+    return_data.workflow_runs_data = localData.workflow_runs_data;
+    return return_data;
+  }
+
+  try {
+    // get latest workflow runs for main branch
+    const workflows = await _getSingleWorkflow(5937682);
```

Comment:
what is this number?

Reply:
functional tests workflow id, creating a const for it, for more clarity

Reply:
Can you use the workflow name instead?